querydsl-core/src/main/java/com/querydsl/core/support/ReactiveFetchableQueryBase.java

```diff
@@ -0,0 +1,51 @@
+package com.querydsl.core.support;
+
+
+import com.querydsl.core.FetchableQuery;
+import com.querydsl.core.ReactiveFetchable;
+import com.querydsl.core.ResultTransformer;
+import com.querydsl.core.types.SubQueryExpression;
+import reactor.core.publisher.Mono;
+
+/**
+ * {@code FetchableQueryBase} extends the {@link QueryBase} class to provide default
+ * implementations of the methods of the {@link com.querydsl.core.ReactiveFetchable} interface
+ *
+ * @param <T> result type
+ * @param <Q> concrete subtype
+ */
+public abstract class ReactiveFetchableQueryBase<T, Q extends ReactiveFetchableQueryBase<T, Q>>
+        extends QueryBase<Q> implements ReactiveFetchable<T> {
+
+    public ReactiveFetchableQueryBase(QueryMixin<Q> queryMixin) {
+        super(queryMixin);
+    }
+
+    @Override
+    public final Mono<T> fetchFirst() {
+        return limit(1).fetchOne();
+    }
+
+    @Override
+    public Mono<T> fetchOne() {
+        return fetch().singleOrEmpty();
```

Comment:
This actually implements the correct behaviour, make sure to update the Javadoc in the interface.

Reply:
yeah the Javadoc wasn't changed - did this